### ISSUE TITLE
Nested vColl elements

### ISSUE THREAD

**Issue by somelinguist** [TEIC/TEI#2034]
The chapter on Feature Structures in the Guidelines says that `vColl` can be nested inside of another `vColl`:

https://github.com/TEIC/TEI/blob/c7dbad9586594427b6068fd8c7f5ae0648db21a7/P5/Source/Guidelines/en/FS-FeatureStructures.xml#L654-L660

However, the model definition for `vColl` only allows for `fs` or elements from `model.featureVal.single`, which doesn't allowing for nesting `vColl` directly.

https://github.com/TEIC/TEI/blob/c7dbad9586594427b6068fd8c7f5ae0648db21a7/P5/Source/Specs/vColl.xml#L31-L40

It is possible to nest `vColl` indirectly inside of another `vColl` by wrapping it inside of a `fs`, `vLabel`, or `vAlt`, but it is not clear if that is what the Guidelines describe.

Is the intention of the Guidelines that `vColl` can be nested directly or only indirectly?

I am unsure if this is the right place to ask this, but I thought it also perhaps might be a bug in either the documentation or the model. Thanks.

**Comment by lb42** [comment]
Definitely if the prose and the schema disagree, one or other needs changing.
My reading of the text you quote above is that a nested vColl does not behave in a very useful way, ("such collections pay no attention to the contents of the nested  <gi>vColl</gi> elements") and it might be better not to permit it, but rather to require use of `<vRange>.` But maybe there is some reason for permitting it which I am unaware of.

**Comment by raffazizzi** [comment]
@laurentromary or @lb42 do you know how the ISO standard deals with this issue? Clearly we need to change either the guidelines or the schema. A VF2F subgroup is leaning towards keeping the schema as is (i.e. `vColl` does not directly nest) and adjust the prose, but we should make this as conformant to the ISO spec as possible.

**Comment by laurentromary** [comment]
The ISO standard also says: "A `<vColl>` element may also collect together one or more other `<vColl>` elements, if, for example, one of the members of a set is itself a set, or if two lists are concatenated together. Note that such collections pay no attention to the contents of the nested `<vColl>` elements: if it is desired to produce the union of two sets, the
<vMerge> element discussed below should be used to make a new collection from the two sets." 
It seems very specific, but we should make `<vCol>` recursive to be compliant with this.

**Comment by lb42** [comment]
As far as I am aware the iso spec doesn't change anything in the tei spec. So if as arguably in this case the tei spec is suboptimal the iso spec will be too. It reads as if the prose was adjusted to specify that lazy evaluation of nested vcolls is obligatory and so tho it looks like recursion it actually isn't. Which will confuse everyone. Imho it would be better to allow vColl to contain some sort of vRef element to make explicit what is going on.

**Comment by raffazizzi** [comment]
I think this is ambiguous: "A `<vColl>` element may also **collect together** one or more other `<vColl>` elements" because collecting (to me at least) implies that they're in the same basket, but it doesn't prescribe how the contents of that basket is organized, or if they are contained by other things. A fruit basket collects apples even if some of them are wrapped in a plastic bag.

"`<vColl>` collects together one ore more `<vColl>` elements" is ambiguous because it applies to both the following examples:
```xml
<!-- valid in current schema -->
<vColl>
   <fs>
      <f name="ex1.1">
         <vColl></vColl>
      </f>
      <f name="ex1.2">
         <vColl></vColl>
      </f>
   </fs>
</vColl>
```
```xml
<!-- not valid -->
<vColl>
   <vColl></vColl>
   <vColl></vColl>
</vColl>
```
However, in the next sentence there's a reference to "such **collections** [of `<vColl>` elements] pay no attention [...]". This tells me that the `<vColl>` that are "collect[ed] together" are collections themselves (which is very different from them being *in* a collection). This indicates that they can (should?) be siblings (a collection typically is a superclass of lists, stacks, sets https://en.wikipedia.org/wiki/Collection_(abstract_data_type)).

"such **collections** [of `<vColl>` elements]" more strongly permits a structure such as:
```xml
<!-- not valid -->
<vColl>
   <vColl></vColl>
   <vColl></vColl>
   <f name="ex1">
      <vColl></vColl>
   </f>
</vColl>
```
In short, I think we should adjust the schema to allow `<vColl>` to nest and leave the prose alone (so that we retain 1:1 correspondence between TEI and ISO).

**Comment by martinascholger** [comment]
Council at VF2F agrees that the schema should be adjusted to allow `<vColl>` to nest.

**Comment by lb42** [comment]
Adjusting the schema to match the prose is not such a good idea when (as here) the prose is muddled. It seems to me to be saying that  you can have apples inside a bag, or a bag of apples inside a bag, but that the apples are to be treated differently in the two cases.
